code/obj/machinery/computer/security/riot.dm

```diff
@@ -135,135 +140,199 @@
 				LAGCHECK(LAG_REALTIME)
 
 	proc/unauthorize()
-		if(src.authed)
+		if(!src.authed)
+			return
 
-			logTheThing("station", usr, null, "unauthorized armory access")
-			authed = 0
-			src.ClearSpecificOverlays("screen_image")
-			icon_state = "drawbr"
-			src.UpdateIcon()
+		logTheThing("station", usr, null, "unauthorized armory access")
+		authed = 0
+		authorisations = list()
+		src.ClearSpecificOverlays("screen_image")
+		icon_state = "drawbr"
+		src.UpdateIcon()
 
-			for (var/obj/machinery/door/airlock/D in armory_area)
-				if (D.has_access(access_security))
-					D.req_access = list(access_maxsec)
-				LAGCHECK(LAG_REALTIME)
+		ON_COOLDOWN(src, "deauth", 5 MINUTES)
 
-			if (armory_area)
-				for(var/obj/O in armory_area)
-					if (istype(O,/obj/storage/secure/crate))
-						O.req_access = list(access_maxsec)
-					else if (istype(O,/obj/machinery/vending))
-						O.req_access = list(access_maxsec)
+		for (var/obj/machinery/door/airlock/D in armory_area)
+			if (D.has_access(access_security))
+				D.req_access = list(access_maxsec)
+			LAGCHECK(LAG_REALTIME)
 
-				LAGCHECK(LAG_REALTIME)
+		if (armory_area)
+			for(var/obj/O in armory_area)
+				if (istype(O,/obj/storage/secure/crate))
+					O.req_access = list(access_maxsec)
+				else if (istype(O,/obj/machinery/vending))
+					O.req_access = list(access_maxsec)
+
+			LAGCHECK(LAG_REALTIME)
 
 	proc/print_auth_needed(var/mob/author)
 		if (author)
 			for (var/mob/O in hearers(src, null))
-				O.show_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[author] request accepted. [src.auth_need - src.authorized.len] authorizations needed until Armory is opened.\"</span></span>", 2)
+				O.show_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[author] request accepted. [src.auth_need - src.authorisations.len] authorizations needed until Armory is opened.\"</span></span>", 2)
 		else
 			for (var/mob/O in hearers(src, null))
-				O.show_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[src.auth_need - src.authorized.len] authorizations needed until Armory is opened.\"</span></span>", 2)
+				O.show_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[src.auth_need - src.authorisations.len] authorizations needed until Armory is opened.\"</span></span>", 2)
+
+	proc/checkRequirements(var/mob/user, var/obj/item/W)
+		// Reset the variables.
+		hasID = FALSE
+		hasAccess = FALSE
+		mismatchedAppearance = FALSE
+		mismatchedID = FALSE
+		maxSecAccess = FALSE
+		existingAuthorisation = FALSE
+		nameOnFile = FALSE
+		printOnFile = FALSE
+
+		if (W == null) // No ID.
+			return
+
+		if (!istype(W, /obj/item/card/id)) // No ID.
+			return
+		else
+			hasID = TRUE
+
+		var/obj/item/card/id/ID = W
+
+		if (!ID:access) // No access.
```

Comment:
```suggestion
		if (!ID.access) // No access.
```
You just cast this to a type that has the property, no need for the cursed `:` (same for all the other `:` usages here)